nixos/modules/services/networking/firewall.nix

```diff
@@ -337,6 +456,121 @@ in
+    };
+
+    networking.firewall.rules = mkOption {
+      default = [];
+      example = [];
+      type = types.listOf types.optionSet;
+      description = ''
+        Rules for the iptables firewall.
+      '';
+      options = {
+        ipv4Only = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Force this rule to apply to ipv4 tables only.
+            Otherwise its decided upon the ipaddress and its format if it
+            applies to ipv6 or not.
+          '';
+        };
+        ipv6Only = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Force this rule to apply to ipv6 tables only.
+            Otherwise its decided upon the ipaddress and its format if it
+            applies to ipv4 or not.
+          '';
+        };
+
+        sourceAddr = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+        destinationAddr = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = ''
+          '';
+        };
+
+        sourcePort = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "10000:15000";
+          description = ''
+            A port or portrange (as string) for the --sport option of iptables.
+            TODO: Validate this on protocol in ["tcp" "udp"]
+          '';
+        };
+        destinationPort = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "80:443";
+          description = ''
+            A port or portrange (as string) for the --dport option of iptables.
+            TODO: Validate this on protocol in ["tcp" "udp"]
+          '';
+        };
+
+        fromInterface = mkOption {
+          default = null;
+          type = types.nullOr types.str;
+          description = ''
+            Source interface to match on.
+            Value is the name of the interface, optionally prefixed with ! to
+            match everything _except_ that interface.
+            Wildcards are allowed: eth+ matches all eth-something interfaces.
+            Special value nil equals all interfaces.
+          '';
+        };
+        toInterface = mkOption {
+          default = null;
+          type = types.nullOr types.str;
+          description = ''
+            Source interface to match on.
+            Value is the name of the interface, optionally prefixed with ! to
+            match everything _except_ that interface.
+            Wildcards are allowed: eth+ matches all eth-something interfaces.
+            Special value nil equals all interfaces.
+          '';
+        };
+        target = mkOption {
+          default = null;
+          type = types.nullOr (
+            types.addCheck types.str (
+              v: v == "ACCEPT" || v == "DROP" || v == "REJECT"
```

Comment:
yeah, enum sounds like the word I didn't remember when writing this ;-)

`LOG` would be the action to just log, nothing else, right?

Also I think having DROP and REJECT both could be seen as superfluous as the actual action depends on the global `rejectPackets` option. (Current difference between DROP and REJECT is that the first sends to nixos-fw-refuse while the later sends to nixos-fw-log-refuse.
Maybe these choices here shouldn't be these abstract names but the real chain names (without the nixos-fw-prefix)?